07_physical_distributions_stopp/02_limnosat_viz.R

```diff
@@ -0,0 +1,334 @@
+library(tidyverse)
+library(leaflet)
+library(sf)
+library(feather)
+library(lubridate)
+library(leafgl)
+library(mapview)
+library(rayshader)
+library(elevatr)
+library(ggridges)
+library(png)
+library(grid)
+library(xkcd)
+
+mapviewOptions(fgb=F)
+
+setwd('07_physical_distributions_stopp/')
+
+## To download the data, this may time out.  If it does, data can be accessed at 
+## https://doi.org/10.5281/zenodo.4139694
+#source('01_limnosat_download.R')
+
+## Read in the data
+# Lakes
+lakes <- st_read('data_in/HydroLakes_DP.shp') %>%
+  filter(type=='dp') %>%
+  st_centroid() %>%
+  st_transform(5070)
+
+#Read in Limnosat
+ls <- read_csv('data_in/LimnoSat_20200628.csv')
+
+##Filter it down a bit to make it less unweidly
+ls <- ls %>% mutate(month = month(date),
+                    doy=yday(date)) %>%
+  filter(pCount_dswe3 == 0,
+         pCount_dswe1 > 9,
+         dWL>470,
+         dWL<584)
+
+#Make a grid of the US to group lakes into
+usa <- maps::map('usa', plot = F) %>% 
+  st_as_sf() %>%
+  st_transform(5070)
+
+mapview(usa)
+
+grid <- st_make_grid(usa, cellsize = c(50000,50000), square = F) %>% st_as_sf() %>% mutate(grid_ID = row_number())
+grid <- st_join(grid,usa,left=F)
+
+mapview(grid,zcol='grid_ID')
+
+grid_lake_walk <- grid %>% st_join(lakes) 
+  st_set_geometry(NULL)
+
+## Lets putz around with Elevatr
+grid_xy <- st_centroid(grid)
+library(elevatr)
+elev_rast <- get_elev_raster(grid_xy,z=3)
+
+elev_rast <- raster::crop(elev_rast, st_bbox(usa))
+elev_rast <- raster::mask(elev_rast,st_buffer(usa,1e5))
+#raster::plot(elev_rast)
+elev_matrix <- raster_to_matrix(elev_rast)
+elev_matrix[elev_matrix < -500] = NA
+attr(elev_matrix,'crs') <- attr(elev_rast,'crs')
+attr(elev_matrix,'extent') <- attr(elev_rast,'extent')
+
+## Make our base layer
+elev_matrix %>%
+  sphere_shade(texture='desert') %>%
+  add_shadow(ray_shade(elev_matrix,zscale=50),0.3) %>%
```

Comment:
Now stuck here. I get the lovely error

```
Error in hillshade[, , 1] * scales::rescale(shadowmap, c(max_darken, 1)) :                                                    
  non-conformable arrays
```

which [I have asked about before](https://github.com/tylermorganwall/rayshader/issues/46) and don't really understand in this context (all the matrices appear to be the same dimensions - 657x405)

Reply:
What version `rayshader` are you using?  I think thats a bug that was fixed more recently than the issue you linked to (https://github.com/tylermorganwall/rayshader/issues/176) , but you may need to do devtools install from github.
`devtools::install_github("tylermorganwall/rayshader")`

Reply:
Good point! I installed `0.24.10` from CRAN today. I see `0.27.4` on github and will try with that version.

Reply:
Success with 0.27.4!